### PR TITLE
`getdlcoffer` RPC

### LINKED ISSUE #4164: Add `getoffer temporaryContractId` endpoint
This allows the frontend to query the backend for an arbitrary `temporaryContractId`. 

The use case is for users to be able to copy the offer tlv _after_ it's initially displayed to the user. Currently if you create an offer via the UI, and then click away without copying the tlv to your clipboard, you have no ability to get the tlv for the offer again.

### PR DESCRIPTION
**Pull request by rorp**
Fixes https://github.com/bitcoin-s/bitcoin-s/issues/4164

### CODE REVIEW

**Comment by Christewart** [comment]
Does not compile

File: app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala [hunk 1]
```diff
@@ -256,6 +256,29 @@ case class WalletRoutes(wallet: AnyDLCHDWalletApi)(implicit
...
+              dlcOpt <- wallet.findDLCByTemporaryContractId(tempContractId)
+              dlc = dlcOpt.getOrElse(
+                throw new IllegalArgumentException(
+                  s"Cannot find a DLC with temp contact ID $tempContractId"))
+              offerOpt <- wallet.getDLCOffer(dlc.dlcId)
+              offer = offerOpt.getOrElse(
+                throw new IllegalArgumentException(
+                  s"Cannot find an offer with for DLC ID ${dlc.dlcId}"))
+            } yield {
+              val tlv = offer.toTLV
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala]: We return the `LnMessage(tlv)` here. We should probably be consistent and return the same format here too.

https://github.com/bitcoin-s/bitcoin-s/blob/9cfe65f1cca8e3a7df78e0c1d07866bfbe9a28da/app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala#L348

**Review by Christewart — CHANGES_REQUESTED** [review]
A unit test would be nice so we can detect future regressions if we ever accidentally break the API

**Review by Christewart — APPROVED** [review]

**Review by user411 — APPROVED** [review]
This tests well for me

**Pull request merged**